### PR TITLE
community: fix Cypher query generation for node names with spaces

### LINKED ISSUE #28375: GraphCypherQAChain Issue with Node Names Containing Spaces
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

from langchain.chains import GraphCypherQAChain
chain = GraphCypherQAChain.from_llm(graph=graph, llm=llm, verbose=True, allow_dangerous_requests=True)
response = chain.invoke({"What is Lana Fathima current  Medical Condition "})

### Error Message and Stack Trace (if applicable)

 {code: Neo.ClientError.Statement.SyntaxError} {message: Invalid input 'Condition': expected a parameter, '&', ')', ':', 'WHERE', '{' or '|' (line 2, column 70 (offset: 76))
"MATCH (p:Person {id: 'Lana Fathima'})-[:DIAGNOSED_WITH]->(mc:Medical Condition)"

### Description

If the node Name is having space in between the genrated cypher query should have
 MATCH (p:Person {id: 'Lana Fathima'})-[:DIAGNOSED_WITH]->(mc:'Medical Condition')
 rather it is genration cypher query "MATCH (p:Person {id: 'Lana Fathima'})-[:DIAGNOSED_WITH]->(mc:Medical Condition)"
 hence it is throughting syntax error

### System Info

System Information
------------------
> OS:  Linux
> OS Version:  #1 SMP PREEMPT_DYNAMIC Fri Jul 12 20:45:27 UTC 2024
> Python Version:  3.9.20 (main, Sep  9 2024, 00:00:00) 
[GCC 11.5.0 20240719 (Red Hat 11.5.0-2)]

Package Information
-------------------
> langchain_core: 0.3.18
> langchain: 0.3.7
> langchain_community: 0.3.7
> langsmith: 0.1.143
> langchain_experimental: 0.3.3
> langchain_huggingface: 0.1.2
> langchain_milvus: 0.1.7
> langchain_ollama: 0.2.0
> langchain_openai: 0.2.9
> langchain_text_splitters: 0.3.2
> langchainhub: 0.1.21
> langgraph: 0.2.52

Optional packages not installed
-------------------------------
> langserve

Other Dependencies
------------------
> aiohttp: 3.11.0
> async-timeout: 4.0.3
> dataclasses-json: 0.6.7
> httpx: 0.27.2
> httpx-sse: 0.4.0
> huggingface-hub: 0.26.2
> jsonpatch: 1.33
> langgraph-checkpoint: 2.0.4
> langgraph-sdk: 0.1.36
> numpy: 1.26.4
> ollama: 0.3.3
> openai: 1.54.4
> orjson: 3.10.11
> packaging: 24.2
> pydantic: 2.9.2
> pydantic-settings: 2.6.1
> pymilvus: 2.4.9
> PyYAML: 6.0.2
> requests: 2.32.3
> requests-toolbelt: 1.0.0
> sentence-transformers: 3.3.1
> SQLAlchemy: 2.0.35
> tenacity: 9.0.0
> tiktoken: 0.8.0
> tokenizers: 0.20.3
> transformers: 4.46.3
> types-requests: 2.32.0.20241016
> typing-extensions: 4.12.2

### PR DESCRIPTION
**Pull request by chkaty**
**Description:**  
This PR fixes an issue in `GraphCypherQAChain` where node names containing spaces (e.g., "Medical Condition") resulted in a `Neo.ClientError.Statement.SyntaxError` when generating Cypher queries. The issue was caused by node names with spaces not being wrapped in backticks, which are required for correct Cypher syntax.
- Modified the `extract_cypher` function to wrap node names containing spaces in backticks.

**Issue:**  
Fixes #28375

**Dependencies:**  N/A

**Test Coverage:**  
- Unit tests have been added to ensure that node names with single and multiple spaces are correctly wrapped in backticks. Additionally, queries with already backtick-wrapped node names remain unaffected.

**Lint and Test:**  
- Ran `make format`, `make lint`, and `make test` to ensure all tests pass and the code adheres to the formatting standards.

**Twitter handle:**  N/A

### CODE REVIEW

**Comment by efriis** [comment]
pinging this over to the neo4j folks because this feels like the kind of thing that could have some sql injection vulnerabilities in it (either existing or new). Will take a look after!

**Comment by efriis** [comment]
Hey there! This PR is against a deprecated community integration that has moved to a dedicated integration package.

I'll close it, and feel free to reopen against the integration package! https://github.com/langchain-ai/langchain-neo4j

**Comment by chkaty** [comment]
Got it, I'll take a look at the dedicated integration package and consider raising this there if necessary. I appreciate your direction on this!

File: libs/community/langchain_community/chains/graph_qa/cypher.py [hunk 1]
```diff
@@ -65,7 +65,17 @@ def extract_cypher(text: str) -> str:
     # Find all matches in the input text
     matches = re.findall(pattern, text, re.DOTALL)
 
-    return matches[0] if matches else text
+    # Extracted Cypher query
+    cypher_query = matches[0] if matches else text
+
+    # Replace node labels with backticks if they contain spaces
+    cypher_query = re.sub(
```
Comment (jexp) [inline libs/community/langchain_community/chains/graph_qa/cypher.py:72]: we can simplify it be always quoting all labels, rel-types and property keys and removing backticks from them.

**Pull request closed without merging**